src/lib/services/api-token-service.ts

```diff
@@ -97,66 +96,21 @@ export class ApiTokenService {
     }
 
     /**
-     * Executed by a scheduler to refresh all active tokens
+     * Called by a scheduler without jitter to refresh all active tokens
      */
     async fetchActiveTokens(): Promise<void> {
         try {
             this.activeTokens = await this.store.getAllActive();
-            this.initialized = true;
-        } finally {
-            // biome-ignore lint/correctness/noUnsafeFinally: We ignored this for eslint. Leaving this here for now, server-impl test fails without it
-            return;
+        } catch (e) {
+            this.logger.warn('Failed to fetch active tokens', e);
         }
     }
 
     async getToken(secret: string): Promise<IApiToken> {
         return this.store.get(secret);
     }
 
-    async updateLastSeen(): Promise<void> {
-        if (this.lastSeenSecrets.size > 0) {
-            const toStore = [...this.lastSeenSecrets];
-            this.lastSeenSecrets = new Set<string>();
-            await this.store.markSeenAt(toStore);
-        }
-    }
-
-    public async getAllTokens(): Promise<IApiToken[]> {
-        return this.store.getAll();
-    }
-
-    public async getAllActiveTokens(): Promise<IApiToken[]> {
-        if (this.flagResolver.isEnabled('useMemoizedActiveTokens')) {
-            if (!this.initialized) {
-                // unlikely this will happen but nice to have a fail safe
-                this.logger.info('Fetching active tokens before initialized');
-                await this.fetchActiveTokens();
-            }
-            return this.activeTokens;
-        } else {
-            return this.store.getAllActive();
-        }
-    }
-
-    private async initApiTokens(tokens: ILegacyApiTokenCreate[]) {
-        const tokenCount = await this.store.count();
-        if (tokenCount > 0) {
-            return;
-        }
-        try {
-            const createAll = tokens
-                .map(mapLegacyTokenWithSecret)
-                .map((t) => this.insertNewApiToken(t, SYSTEM_USER_AUDIT));
-            await Promise.all(createAll);
-        } catch (e) {
-            this.logger.error('Unable to create initial Admin API tokens');
-        }
-    }
-
-    public async getUserForToken(
-        secret: string,
-        flagContext?: IFlagContext, // temporarily added, expected from the middleware
-    ): Promise<IApiUser | undefined> {
+    async getTokenWithCache(secret: string): Promise<IApiToken | undefined> {
```

Comment:
This method is extracted from `getUserForToken` it was moved up to be close to `getToken` method above